crates/biome_js_analyze/src/lint/nursery/use_jsx_curly_brace_convention.rs

```diff
@@ -0,0 +1,348 @@
+    AddBraces,
+    /// The user should remove the curly braces around the expression.
+    RemoveBraces,
+}
+
+impl Rule for UseJsxCurlyBraceConvention {
+    type Query = Ast<UseJsxCurlyBraceConventionQuery>;
+    type State = CurlyBraceResolution;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let query = ctx.query();
+        let has_curly_braces = has_curly_braces(query);
+        match query {
+            UseJsxCurlyBraceConventionQuery::JsxAttributeInitializerClause(attr) => {
+                handle_attr_init_clause(attr, has_curly_braces)
+            }
+            UseJsxCurlyBraceConventionQuery::AnyJsxChild(child) => {
+                handle_jsx_child(child, has_curly_braces)
+            }
+        }
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        let source_range = match &node {
+            UseJsxCurlyBraceConventionQuery::JsxAttributeInitializerClause(node) => {
+                node.value().map(|value| value.range())
+            }
+            _ => Ok(node.range()),
+        }
+        .unwrap_or(node.range());
+
+        let diag = match (state, node) {
+            (CurlyBraceResolution::AddBraces, UseJsxCurlyBraceConventionQuery::JsxAttributeInitializerClause(_)) => RuleDiagnostic::new(
+                rule_category!(),
+                source_range,
+                markup! {
+                    "Should have curly braces around expression."
+                },
+            )
+            .note(markup! {
+                "JSX attribute value should be wrapped in curly braces. This will make the JSX attribute value more readable."
+            }),
+            (CurlyBraceResolution::AddBraces, UseJsxCurlyBraceConventionQuery::AnyJsxChild(_)) => RuleDiagnostic::new(
+                rule_category!(),
+                source_range,
+                markup! {
+                    "Should have curly braces around expression."
+                },
+            )
+            .note(markup! {
+                "JSX child should be wrapped in curly braces."
+            }),
+            (CurlyBraceResolution::RemoveBraces, UseJsxCurlyBraceConventionQuery::JsxAttributeInitializerClause(_)) => RuleDiagnostic::new(
+                rule_category!(),
+                source_range,
+                markup! {
+                    "Should not have curly braces around expression."
+                },
+            )
+            .note(markup! {
+                "JSX attribute value does not need to be wrapped in curly braces."
+            }),
+            (CurlyBraceResolution::RemoveBraces, UseJsxCurlyBraceConventionQuery::AnyJsxChild(_)) => RuleDiagnostic::new(
+                rule_category!(),
+                source_range,
+                markup! {
+                    "Should not have curly braces around expression. "
```

Comment:
Maybe you can remove the space?
`"Should not have curly braces around expression. "` -> `"Should not have curly braces around expression."`